code/game/objects/items/storage/pouch.dm

```diff
@@ -593,9 +592,26 @@
 	else
 		return ..()
 
-/obj/item/storage/pouch/flare/full
-	fill_type = /obj/item/explosive/grenade/flare
-	fill_number = 7
+/obj/item/storage/pouch/flare/attackby_alternate(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/weapon/gun/grenade_launcher/single_shot/flare))
+		return ..()
+	var/obj/item/weapon/gun/grenade_launcher/single_shot/flare/flare_gun = I
+	for(var/obj/item/flare in contents)
+		if(!(flare.type in can_hold))
+			continue
+		if(user.l_hand && user.r_hand || length(flare_gun.chamber_items))
+			flare_gun.tactical_reload(flare, user)
+		else
+			flare_gun.reload(flare, user)
+		orient2hud()
+		return
+
+/obj/item/storage/pouch/flare/full/Initialize()
+	. = ..()
+	if(max_storage_space)
+		for(var/i in 1 to (max_storage_space - 1))
```

Comment:
this will overload the pouch. The pouch has 28 space, and you're filling it up with 27 worth of flares. The flaregun is 2-3, so won't fit. the storage_type_limit just overrides the size limit, but the max space limit must be obeyed.

Reply:
Nah, the flare gun was already a tiny item. It's not overloading the pouch.

Reply:
It's fucking tiny? Guh, who did that. Means you can fit two in your hat.

Reply:
Incredibly funny.